cluster/src/main/java/org/apache/iotdb/cluster/qp/callback/MultiQPTask.java

```diff
@@ -28,19 +29,20 @@
   Map<String, QPTask> taskMap;
 
   /**
-   * Task thread map
+   * Each future task handle a request in taskMap, which is corresponding to a group id. String:
+   * group id
    */
-  Map<String, Thread> taskThreadMap;
+  Map<String, Future<?>> taskThreadMap;
 
-  public MultiQPTask(boolean isSyncTask, int taskNum, TaskState taskState, TaskType taskType) {
+  public MultiQPTask(boolean isSyncTask, int taskNum, TaskType taskType) {
     super(isSyncTask, taskNum, TaskState.INITIAL, taskType);
   }
 
   @Override
   public void shutdown() {
-    for (Thread taskThread : taskThreadMap.values()) {
-      if (taskThread.isAlive() && !taskThread.isInterrupted()) {
-        taskThread.interrupt();
+    for (Future<?> task : taskThreadMap.values()) {
+      if (task.isDone()) {
```

Comment:
if task is done, why you cancel it?
else take is not done, you do nothing?

Reply:
fixed